src/strategies/BeefyAeroStrategy.sol

```diff
@@ -0,0 +1,313 @@
+
+import {IAeroRouter, IAeroPool} from "src/interfaces/IAerodrome.sol";
+import {IBeefyVault} from "src/interfaces/Beefy.sol";
+
+import {AccessStrategy} from "src/strategies/AccessStrategy.sol";
+
+import {AffineVault} from "src/vaults/Vault.sol";
+import {StrategyVault} from "src/vaults/locked/StrategyVault.sol";
+
+contract BeefyAeroStrategy is AccessStrategy {
+    using FixedPointMathLib for uint256;
+    using SafeTransferLib for ERC20;
+
+    // beefy vault
+    IBeefyVault public immutable beefy;
+    // const max bps
+    uint256 public constant MAX_BPS = 10_000;
+    uint256 public defaultSlippageBps;
+
+    ERC20 public immutable token0;
+    ERC20 public immutable token1;
+
+    IAeroRouter public immutable aeroRouter;
+    IAeroPool public immutable lpToken;
+
+    constructor(
+        AffineVault _vault,
+        IBeefyVault _beefy,
+        IAeroRouter _router,
+        ERC20 _token1,
+        address[] memory strategists
+    ) AccessStrategy(_vault, strategists) {
+        beefy = _beefy;
+        token0 = asset;
+        token1 = _token1;
+
+        aeroRouter = _router;
+        lpToken = IAeroPool(aeroRouter.poolFor(address(token0), address(token1), false, aeroRouter.defaultFactory()));
+
+        require(address(beefy.want()) == address(lpToken), "BAS: Invalid beefy vault asset.");
+
+        // approve assets to use in aerodrome router
+        token0.approve(address(aeroRouter), type(uint256).max);
+        token1.approve(address(aeroRouter), type(uint256).max);
+
+        // approve lp token for pearlRouter and beefy vault
+        lpToken.approve(address(aeroRouter), type(uint256).max);
+        lpToken.approve(address(beefy), type(uint256).max);
+
+        // @dev check for valid token0 and token1 from pool as poolFor generate the address
+        (address t0, address t1) = aeroRouter.sortTokens(address(token0), address(token1));
+        require(lpToken.token0() == t0, "BAS: invalid token0.");
+        require(lpToken.token1() == t1, "BAS: invalid token1.");
+    }
+
+    /**
+     * @notice calculate min/max amount after slippage
+     * @param amount token amount
+     * @param slippageBps slippage
+     * @param isMin return min amount after slippage, otherwise return max amount
+     */
+    function _calculateSlippageAmount(uint256 amount, uint256 slippageBps, bool isMin)
+        internal
+        pure
+        returns (uint256)
+    {
+        if (isMin) {
+            return amount.mulDivDown(MAX_BPS - slippageBps, MAX_BPS);
+        }
+        return amount.mulDivDown(MAX_BPS, MAX_BPS - slippageBps);
+    }
+
+    /**
+     * @notice return swapped token amount
+     * @param from token to swap
+     * @param to token swapped into
+     * @param amount amount of from token
+     */
+    function _getSwapPrice(ERC20 from, ERC20 to, uint256 amount) internal view returns (uint256 tokenToAmount) {
+        IAeroRouter.Route[] memory route = new IAeroRouter.Route[](1);
+
+        route[0].from = address(from);
```

Comment:
You can set route 0 to a struct value directly instead of setting the 4 fields individually

```
route[0] = Route({from: from, to: to, stable: false, factory : factory})
```

Reply:
done